debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/pointer/helper/PrintObjectFieldsHelper.java

```diff
@@ -123,24 +125,31 @@ public static void printJ9ObjectFields(PrintStream out, int tabLevel, J9ClassPoi
 				clazz = nestedClassHierarchy[i];
 				depth = J9ClassHelper.classDepth(clazz).longValue();
 				boolean found = false;
-				for (superclassIndex = 0; (superclassIndex <= depth) && !found; superclassIndex++) {
-					J9ClassPointer superclass;
-					if (superclassIndex == depth) {
-						superclass = clazz;
-					} else {
-						superclass = J9ClassPointer.cast(clazz.superclasses().at(superclassIndex));
-					}
-					Iterator<J9ObjectFieldOffset> iterator = J9ObjectFieldOffsetIterator.J9ObjectFieldOffsetIteratorFor(superclass.romClass(), clazz, previousSuperclass, flags);
-
-					while (iterator.hasNext()) {
-						J9ObjectFieldOffset result = iterator.next();
-						if (result.getName().equals(nestingHierarchy[i])) {
-							nestedFieldOffset = nestedFieldOffset.add(result.getOffsetOrAddress());
-							found = true;
-							break;
+
+				if (J9ClassHelper.isArrayClass(clazz)) {
+					int index = Integer.parseInt(nestingHierarchy[0].substring(1, nestingHierarchy[0].length() - 1));
```

Comment:
Where is the check that the string `nestingHierarchy[0]` is of the form `[\d+]` so we know we're parsing something reasonable?

Reply:
This function is only called if flatobject is true and the condition for flatobject is `((null != nestingHierarchy) && (nestingHierarchy.length > 0))`

Reply:
So at this point, would it be better to have a check and then throw an IllegalArgumentException if it's not a in form of an int?

Reply:
This sort of impedance mismatch makes me wonder whether it's time to replace
`String[] nestingHierarchy` with a better context representation.

Reply:
Rather, what I could do is, instead of calling `extraArgs[0].startsWith("[")` like you suggested, I could do the Pattern match at that very top level before the PrintObjectFields function is even called and then just display an error message there if the arguments are ill-formatted. At this point, if the `nestingHierarchy[0]` isn't in the form `[\d]`, our earlier call on `findNestedClassHierarchy` would not have executed the appropriate logic since the pattern match done in the ValueTypeHelper.java would have returned false.